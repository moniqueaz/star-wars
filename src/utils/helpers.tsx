export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
