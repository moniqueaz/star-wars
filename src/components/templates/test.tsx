import { screen } from '@testing-library/react';
import React from 'react';
import { renderWithTheme } from 'utils/tests/helpers';

import Template from '.';

type ContainerProps = {
  children: React.ReactNode
}

jest.mock('components/atoms/Container', () => {
  return {
    Container: function Mock( { children }: ContainerProps) {
      return <div data-testid="Mock Container">{children}</div>;
    },
  };
} );

describe('<Template />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Template><div data-testid="Template">Template</div></Template>);
    expect(screen.getByTestId('Template')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );
} );
