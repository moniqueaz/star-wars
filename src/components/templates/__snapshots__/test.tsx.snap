// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> should render the heading 1`] = `
.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-image: linear-gradient(to bottom,#1A4173,#0C1D33);
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4.8rem;
}

.c1:before {
  content: '';
  background-image: linear-gradient(to right,transparent,#fff,transparent);
  height: 2px;
  width: 100%;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  opacity: .5;
  z-index: 0;
  -webkit-animation: show 1s linear;
  animation: show 1s linear;
}

.c2 {
  width: 15rem;
  height: auto;
}

.c4 {
  text-align: center;
  background-color: #333;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4.6rem;
}

.c5 {
  width: 100%;
  color: #f1f1f1;
  font-size: 1.6rem;
}

.c0 {
  font-size: 1.6rem;
}

.c3 {
  min-height: calc(100vh - 2rem - 4.8rem - 4.6rem);
}

@media (min-width:1024px) {
  .c1 {
    height: 14rem;
  }
}

@media (min-width:1024px) {
  .c1:before {
    height: 3px;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 25rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    height: 7.6rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    min-height: calc(100vh - 2rem - 14rem - 7.6rem);
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="logo star wars"
      class="c2"
      src="/img/logo.png"
    />
  </div>
  <div
    data-testid="Mock Container"
  >
    <div
      class="c3"
    >
      <div
        data-testid="Template"
      >
        Template
      </div>
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      data-testid="Mock Container"
    >
      <p
        class="c5"
      >
        Star Wars Card Game
      </p>
    </div>
  </div>
</div>
`;
